flink-sql-runner/src/main/java/com/github/streamshub/flink/KubernetesSecretInterpolator.java

```diff
@@ -0,0 +1,56 @@
+package com.github.streamshub.flink;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class KubernetesSecretInterpolator implements Interpolator {
+    private final KubernetesClient client;
+
+    // Expected pattern for a secret is {{secret:<namespace>/<name>/<key>}}
+    private static final Pattern SECRET_PATTERN = Pattern.compile("\\{\\{secret:([^/]+)/([^}]+)/([^}]+)}}");
+
+    public KubernetesSecretInterpolator() {
+        this.client = new KubernetesClientBuilder().build();
+    }
+
+    // visible for testing
+    public KubernetesSecretInterpolator(KubernetesClient client) {
```

Comment:
Could this be protected rather than public? I also quite like the Strimzi convention of prepending methods like this with `/* test */` rather than having the comment on a separate line